returnn/oggzip.py

```diff
@@ -1,104 +1,200 @@
+        """
+        self.bliss_corpus = bliss_corpus
+        self.segments = segments
+        self.rasr_cache = rasr_cache
+        self.raw_sample_rate = raw_sample_rate
+        self.feat_sample_rate = feat_sample_rate
+        self.no_conversion = no_conversion
+        self.no_audio = no_audio
+        self.concurrent = (
+            len(segments.hidden_paths) if isinstance(segments, MultiOutputPath) else 1
+        )
+
+        self.returnn_python_exe = (
+            returnn_python_exe
+            if returnn_python_exe is not None
+            else gs.RETURNN_PYTHON_EXE
+        )
+        self.returnn_root = (
+            returnn_root if returnn_root is not None else gs.RETURNN_ROOT
+        )
+
+        self.zip_subarchives = (
+            MultiOutputPath(
+                self,
+                "ogg.$(TASK).zip",
+                {i: "ogg.{}.zip".format(i) for i in range(1, self.concurrent + 1)},
+                cached=True,
+            )
+            if self.concurrent > 1
+            else None
+        )
+
+        self.out_ogg_zip = self.output_path("out.ogg.zip")
+
+        self.rqmt = None
+
+    def tasks(self):
+        if self.rqmt:
+            yield Task("run", rqmt=self.rqmt, args=range(1, self.concurrent + 1))
+        else:
+            yield Task("run", mini_task=True, args=range(1, self.concurrent + 1))
+        if self.concurrent > 1:
+            yield Task("merge", mini_task=True)
+
+    def run(self, task_id):
+        output = (
+            self.zip_subarchives.hidden_paths[task_id]
+            if self.concurrent > 1
+            else "out.ogg.zip"
+        )
+        args = [
+            tk.uncached_path(self.returnn_python_exe),
+            os.path.join(
+                tk.uncached_path(self.returnn_root), "tools/bliss-to-ogg-zip.py"
+            ),
+            tk.uncached_path(self.bliss_corpus),
+            "--output",
+            output,
+        ]
+        if isinstance(self.segments, MultiOutputPath):
+            args.extend(["--subset_segment_file", self.segments.hidden_paths[task_id]])
+        elif self.segments is not None:
+            args.extend(["--subset_segment_file", tk.uncached_path(self.segments)])
+
+        if self.no_audio:
+            args.extend(["--no_ogg"])
+        elif self.no_conversion:
+            args.extend(["--no_conversion"])
+        else:
+            if self.rasr_cache is not None:
+                args.extend(["--sprint_cache", tk.uncached_path(self.rasr_cache)])
+            if self.raw_sample_rate is not None:
+                args.extend(["--raw_sample_rate", str(self.raw_sample_rate)])
+            if self.feat_sample_rate is not None:
+                args.extend(["--feat_sample_rate", str(self.feat_sample_rate)])
+
+        sp.check_call(args)
+        if self.concurrent == 1:
+            relink("out.ogg.zip", self.out_ogg_zip.get_path())
+
+    def merge(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:

```

Comment:
```suggestion
        with tempfile.TemporaryDirectory(prefix=gs.TMP_PREFIX) as tmp_dir:
```


see also https://github.com/rwth-i6/i6_core/pull/203
The location where the temporary dictionary is created depends on a couple of factors an is mostly controlled by environment variables. But those are read before Sisyphus cleans up or sets the env variables. With this change the folder will be created by a sisyphus controlled location.

Reply:
> if you use /local you should have more space available, which is 180GB, so you could use that..

With this change, you can specify in your settings e.g.
```
TMP_PREFIX = "/local/sis_"
```
and the local partition will be used.

Reply:
and the default env variable TMPDIR can remain the same (for example for the cluster)

at the chair you have to clean `/local` manually. for single-user desktop this might not be a problem for multi-user this could be an issue..

normally compute or io heavy jobs should be submitted to the cluster then you also don't need another variable in the `settings.py` which might cause problems if somebody uses the Job without the global settings variable

Reply:
lange rede kurzer sinn: easiest and safest would probably be just to submit to the cluster xD

Reply:
>you have to clean `/local` manually.

I understand this happens automatically as soon as the the `with ...` block ends.

> you also don't need another variable in the settings.py

This variable already exists. cf. https://github.com/rwth-i6/sisyphus/blob/fe8ad9fb112672546ec2fc4f180df7610c88b372/sisyphus/global_settings.py#L272
We would only have to overwrite it, if we don't want to use the default location.

Reply:
> I understand this happens automatically as soon as the the with ... block ends.

What if the job crashes??

ah I see. so setting `TMP_PREFIX` in `settings.py` would influence all other Jobs as well, or not? But we don't want different behaviour for cluster and desktop machines.

cluster: `/var/tmp` lots of space
desktop: `/local` "lots" of space

so we want to keep `/var/tmp` as temp location but only want to change it for one single job.. that seems like a lot of unnecessary effort and potential for errors compared to just submitting to cluster :)

Reply:
> safest would probably be just to submit to the cluster

I don't disagree. But I think still the suggestion should be added to add the configurability and to match the sisyohus mktemp mechanism (https://github.com/rwth-i6/sisyphus/blob/fe8ad9fb112672546ec2fc4f180df7610c88b372/sisyphus/toolkit.py#L82-L105)

Reply:
OK. I agree :)

submit to cluster and add the sisyphus mktemp mechanism

Reply:
good point.. I ll check that

Reply:
I added @michelwi's suggestion, also it's submitted to the cluster by default now. You could configure it to be a mini_task, but only from outside the job after initializing it (same as for the existing `run` task).